src/Compilers/CSharp/Portable/Parser/QuickScanner.cs

```diff
@@ -209,7 +211,13 @@ private SyntaxToken QuickScanSyntaxToken()
                 var flags = uc < charPropLength ? (CharFlags)s_charProperties[uc] : CharFlags.Complex;
 
                 state = (QuickScanState)s_stateTransitions[(int)state, (int)flags];
-                if (state == QuickScanState.Done || state == QuickScanState.Bad)
+                // NOTE: that Bad > Done and it is the only state like that
+                // as a result, we will exit the loop on either Bad or Done.
+                // the assert before will validate that these are the only states on which we exit
+                // NOTE: Also note that we must exit on Done or Bad
```

Comment:
**NOTE**: Also **note**

I am not sure if NOTE prefix is useful at all, it used above as well .